lib/Sema/ConstraintSystem.cpp

```diff
@@ -2783,6 +2783,27 @@ void ConstraintSystem::generateConstraints(
   }
 }
 
+Optional<ConstraintSystem::ArgumentInfo>
+ConstraintSystem::getArgumentInfo(const ConstraintLocatorBuilder builder) {
+  Expr *anchor = builder.getAnchor();
+  if (!anchor)
+    return None;
+
+  ConstraintLocator *locator = nullptr;
+  if (auto *apply = dyn_cast<ApplyExpr>(anchor)) {
+    auto *fnExpr = getArgumentLabelTargetExpr(apply->getFn());
+    locator = getConstraintLocator(fnExpr);
+  } else {
+    locator = getCalleeLocator(anchor);
+  }
```

Comment:
This block of code is the same as in `associateArgumentLabels`. Can you pull those out into a method `getArgumentInfoLocator` that we use in both places?

Reply:
Sure! Will do!

Reply:
Done!